jb/ut_book_depth_statistics.cpp

```diff
@@ -0,0 +1,84 @@
+#include <jb/book_depth_statistics.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+constexpr bool eqTypes() {
```

Comment:
That should be eq_types() or equal_types().  Or maybe you can just use a:

    static_assert(
        std::is_same<jb::itch5::book_depth_t, jb::book_depth_t>::value,
        "Mismatched definition of jb::itch5::book_depth_t and jb::book_depth_t");

That static assert can go in the body of any test, it will stop at compile time if the types do not match.